corehq/apps/data_dictionary/views.py

```diff
@@ -89,8 +89,12 @@ def update_case_property(request, domain):
                 name=name, case_type__name=case_type, case_type__domain=domain
             )
         except CaseProperty.DoesNotExist:
-            case_type_obj = CaseType.objects.get(domain=domain, name=case_type)
-            prop = CaseProperty.objects.create(case_type=case_type_obj, name=name)
+            key = 'data-dict-property-{domain}-{type}-{name}'.format(
+                domain=domain, type=case_type, name=name
+            )
+            with CriticalSection([key]):
+                case_type_obj = CaseType.objects.get(domain=domain, name=case_type)
+                prop = CaseProperty.objects.create(case_type=case_type_obj, name=name)
```

Comment:
sorry if i didn't make this clear but the entire get or create code should be within the `CriticalSection`, so everything in the try/except should be under it.  i think it would be better to just write a function named `get_or_create_case_property` which does the enter get or create, since there's another place in this commit that seems to do the same thing